recipes/lzham/all/conanfile.py

```diff
@@ -0,0 +1,221 @@
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, "CMakeLists.txt"),
+                "add_subdirectory(lzhamtest)\n",
+                ""
+            )
+
+            # This line in the root CMakeLists.txt can cause issues, see
+            # https://cmake.org/cmake/help/latest/policy/CMP0077.html.
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, "CMakeLists.txt"),
+                "option(BUILD_SHARED_LIBS \"build shared/static libs\" ON)",
+                ""
+            )
+            return
+        new_sln = []
+        # Remove example and test projects from sln.
+        with open(os.path.join(
+            self.source_folder, "lzham.sln"
+        ), encoding="utf-8") as f:
+            line = f.readline()
+            while line:
+                if (
+                    line.startswith("Project(")
+                    and ("lzhamtest" in line or "example" in line)
+                ):
+                    # Don't write the current line and skip the "EndProject"
+                    # line.
+                    f.readline()
+                else:
+                    new_sln.append(line)
+                line = f.readline()
+        with open(os.path.join(
+            self.source_folder, "lzham.sln"
+        ), "w", encoding="utf-8") as f:
+            f.write("".join(new_sln))
+
+        # Inject conantoolchain.props so that correct platform toolset is used.
+        projects = [(x, f"{x}.vcxproj") for x in (
+            "lzhamcomp",
+            "lzhamdecomp",
+            "lzhamlib",
+        )]
+        projects.append(("lzhamdll", "lzham.vcxproj"))
+        search_str = (
+            '  <Import Project='
+            '"$(VCTargetsPath)\\Microsoft.Cpp.Default.props" />'
+        )
+
+        for p in projects:
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, *p),
+                search_str,
+                '  <ImportGroup Label="PropertySheets">\n'
+                    '    <Import Project="..\\conan\\conantoolchain.props" />\n'
+                    '  </ImportGroup>\n'
+                    + search_str
+            )
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        if is_msvc(self):
+            vs_layout(self)
+        else:
+            cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(
+            self,
+            **self.conan_data["sources"][self.version],
+            destination=self.source_folder,
+            strip_root=True
+        )
+
+    def generate(self):
+        if is_msvc(self):
+            tc = MSBuildToolchain(self)
```

Comment:
Thanks for the review! Part of the reason is that the CMakeLists configure compiler arguments that MSVC does not understand, leading to errors like
```
cl : command line  error D8021: invalid numeric argument '/Wextra' [C:\Users\Babbo\.conan\data\lzham\1.0.0\_\_\build\5a
61a86bb3e07ce4262c80e1510f9c05e9b6d48b\build\lzhamdecomp\lzhamdecomp.vcxproj]
```

Another reason is that I suspect that the MSVC project is "fine-tuned" for Windows and it is the [recommended](https://github.com/richgel999/lzham_codec#compiling-lzham) way to compile for Windows, so I believe it would be less optimal to make a patch adapt the CMakeLists so that MSVC is handled, not to mention a bit of a pain!